docs/src/content/docs/guides/sidebar.mdx

```diff
@@ -0,0 +1,461 @@
+
+## Badges
+
+Links can also include a `badge` property to display a badge next to the link label.
+
+```js
+starlight({
+	sidebar: [
+		{
+			label: 'Guides',
+			items: [
+				// A link with a "New" badge.
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: 'New',
+				},
+			],
+		},
+	],
+});
+```
+
+The above configuration will generate the following sidebar:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: { text: 'New', variant: 'default' },
+				},
+			],
+		},
+	]}
+/>
+
+The `badge` property includes a `text` property (the content to display in the badge) and optionally a `variant` property (to define the type of badge shown).
+
+`badge.text` - a string representing the content to display. (e.g. "New")
+
+`badge.variant` - to override the `default` styling setting, set to one of the following properties for custom styling: `note`, `tip`, `danger`, `caution` or `success`.
+
+```js
+starlight({
+	sidebar: [
+		{
+			label: 'Guides',
+			items: [
+				// A link with a green "New" badge.
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: { text: 'Experimental', variant: 'caution' },
+				},
+			],
+		},
+	],
+});
+```
+
+The above configuration will generate the following sidebar:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: { text: 'Experimental', variant: 'caution' },
+				},
+			],
+		},
+	]}
+/>
+
+## Autogenerated groups
+
+Any group (or sub-group) can be automatically generated, and will be sorted alphabetically by filename. This is helpful when you do not want to manually enter each sidebar item in a group.
+
+To create an autogenerated group, set `autogenerate` to an object specifying the `directory`. For example, given the following file structure:
+
+<FileTree>
+
+- src/
+  - content/
+    - docs/
+      - guides/
+        - components.md
+        - i18n.md
+        - advanced/
+          - project-structure.md
+
+</FileTree>
+
+And the following configuration:
+
+```js
+starlight({
+	sidebar: [
+		{
+			label: 'Guides',
+			// Autogenerate a group of links for the 'guides' directory.
+			autogenerate: { directory: 'guides' },
+		},
+	],
+});
+```
+
+The following sidebar will be generated:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{ label: 'Components', link: '/guides/components/' },
+				{ label: 'Internationalization (i18n)', link: '/guides/i18n/' },
+				{
+					label: 'advanced',
+					items: [
+						{ label: 'Project Structure', link: '/guides/project-structure/' },
+					],
+				},
+			],
+		},
+	]}
+/>
+
+### Customizing autogenerated links in frontmatter
+
+Use the [`sidebar` frontmatter field](/reference/frontmatter/#sidebar) to customize autogenerated links.
```

Comment:
Small suggestion to underline that the frontmatter is not in the Astro config file like most examples so far.

```suggestion
Use the [`sidebar` frontmatter field](/reference/frontmatter/#sidebar) in individual pages to customize autogenerated links.
```